blackjack/core.py

```diff
@@ -0,0 +1,278 @@
+
+
+def render_dealer(x):
+    first_card = "🂠 "
+    ren_hand = ''
+    for card in x[1:]:
+        s = card.suit
+        r = card.rank
+        ren_card = ''
+        if r == Rank.Ace:
+            ren_card = "A"
+        if r == Rank.Two:
+            ren_card = '2'
+        if r == Rank.Three:
+            ren_card = '3'
+        if r == Rank.Four:
+            ren_card = '4'
+        if r == Rank.Five:
+            ren_card = '5'
+        if r == Rank.Six:
+            ren_card = '6'
+        if r == Rank.Seven:
+            ren_card = '7'
+        if r == Rank.Eight:
+            ren_card = '8'
+        if r == Rank.Nine:
+            ren_card = '9'
+        if r == Rank.Ten:
+            ren_card = '10'
+        if r == Rank.Jack:
+            ren_card = "J"
+        if r == Rank.Queen:
+            ren_card = "Q"
+        if r == Rank.King:
+            ren_card = "K"
+        if s == Suit.Spades:
+            ren_card += '♠ '
+            ren_hand += ren_card
+        if s == Suit.Hearts:
+            ren_card += '♥ '
+            ren_hand += ren_card
+        if s == Suit.Diamonds:
+            ren_card += '♦ '
+            ren_hand += ren_card
+        if s == Suit.Clubs:
+            ren_card += '♣ '
+            ren_hand += ren_card
+
+    return first_card + ren_hand
+
+
+def card_point(x):
+    n = x.rank
+    if n == Rank.Ace:
+        return 11
+    if n == Rank.Two:
+        return 2
+    if n == Rank.Three:
+        return 3
+    if n == Rank.Four:
+        return 4
+    if n == Rank.Five:
+        return 5
+    if n == Rank.Six:
+        return 6
+    if n == Rank.Seven:
+        return 7
+    if n == Rank.Eight:
+        return 8
+    if n == Rank.Nine:
+        return 9
+    if n == Rank.Ten:
+        return 10
+    if n == Rank.Jack:
+        return 10
+    if n == Rank.Queen:
+        return 10
+    if n == Rank.King:
+        return 10
+
+
+def get_winner(state: GameState) -> Win:
+    player_total = hand_total(state.player_hand)
+    house_total = hand_total(state.dealer_hand)
+    if player_total == house_total < 21:
+        return Win.Push
+    if player_total == 21 and house_total == 21:
+        return Win.Push
+    if player_total == 21 and house_total != 21:
+        return Win.Player_bj
+    if house_total == 21 and player_total != 21:
+        return Win.House_bj
+    if house_total < 21 < player_total:
+        return Win.Player_bust
+    if house_total > 21 > player_total:
+        return Win.House_bust
+    if player_total < house_total < 21:
+        return Win.House_win
+    if player_total > house_total < 21 and house_total >= 17:
+        return Win.Player_win
+
+
+def parse_play(s: str) -> Optional[Play]:
+    if s == 'y':
+        return Play.Hit
+    if s == 'n':
+        return Play.Stay
+    return None
+
+
+
+# TODO Separate IO and logic
+def has_blackjack(hand: List[Card]):
+    total = hand_total(hand)
+    if total != 21:
+        return False
+    if total == 21:
+        return True
```

Comment:
Redundant check for both `!=` and `==`. When you compare with `==` you already get a bool, so you should just return the result of `== 21`.